### ISSUE TITLE
Bindgen fails to generate constants if `#define`s are out of order

### ISSUE THREAD

**Issue by roypat** [rust-lang/rust-bindgen#2597]
### Input C/C++ Header

```C
#define A B
#define B 1
```

### Bindgen Invocation

```bash
bindgen input.h
```

### Actual Results

```rust
/* automatically generated by rust-bindgen 0.66.1 */

pub const B: u32 = 1;
```

### Expected Results

I expected 

```rust
pub const A: u32 = 1;
pub const B: u32 = 1;
```

Both gcc and clang accept these kind of "out of order" definitions (as counter-intuitive as it is), and the linux kernel makes use of this in `kvm.h`, which is how we discovered this issue in https://github.com/rust-vmm/kvm-bindings/pull/89

**Comment by pvdrz** [comment]
I think https://github.com/rust-lang/rust-bindgen/pull/2561 doesn't strictly fix this but it is a step in that direction iiuc (cc @reitermarkus).

If that's the case maybe this is a good enough reason to do this change, what do you think @emilio?

**Comment by reitermarkus** [comment]
No, https://github.com/rust-lang/rust-bindgen/pull/2561 doesn't fix this, but https://github.com/rust-lang/rust-bindgen/pull/2369 which depends on https://github.com/rust-lang/rust-bindgen/pull/2561 will fix it.
